### ISSUE TITLE
Scripting - TypeError: Cannot read property 'replace' of null

### ISSUE THREAD

**Issue by leathej1** [sitespeedio/browsertime#745]
I am running a blank scrtipt against the 8.0.6 version of the Docker image on MacOS Mojave. My Docker images and my files are located in the same directory:
`docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:8.0.6 blank.js`

The script consists of only:
`module.exports = async function(context, commands) {
  // add your own code here
}`

I am getting the following error:

```[2019-01-31 17:46:20] ERROR: TypeError: Cannot read property 'replace' of null
    at /usr/src/app/node_modules/browsertime/lib/support/pathToFolder.js:40:37
    at Array.forEach (<anonymous>)
    at module.exports (/usr/src/app/node_modules/browsertime/lib/support/pathToFolder.js:39:18)
    at Collector.perIteration (/usr/src/app/node_modules/browsertime/lib/core/engine/collector.js:176:19)
    at Engine.runByScript (/usr/src/app/node_modules/browsertime/lib/core/engine/index.js:118:23)
    at process._tickCallback (internal/process/next_tick.js:68:7)

**Comment by andreicontan** [comment]
Still reproducible with 8.1.1 - is there any workaround until the issue is addressed?

**Comment by soulgalore** [comment]
Hey @andreicontan do you mean the same as @leathej1 case, you are running it with an empty script? What's the use case there, I don't follow? If you get it with a script that tries to do anything, please create a new issue and follow https://www.sitespeed.io/documentation/sitespeed.io/bug-report/ then I'll do my best to help you out :)

Best
Peter

**Comment by andreicontan** [comment]
I'm using the wikipedia login flow part of the documentation with valid credendials, than run from the command line
docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:8.1.1 -b chrome wikipedia-login.js 

I get this in return
` docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:8.0.6 -b chrome wikipedia-login.js 
Google Chrome 71.0.3578.80 
Mozilla Firefox 64.0
[2019-02-04 13:39:39] INFO: Versions OS: linux 4.9.125-linuxkit nodejs: v10.13.0 sitespeed.io: 8.0.6 browsertime: 4.0.5 coach: 3.1.2
[2019-02-04 13:39:39] INFO: Running tests using Chrome - 3 iteration(s)
[2019-02-04 13:39:40] INFO: Start to measure module.exports
[2019-02-04 13:39:41] ERROR: TypeError: Cannot read property 'replace' of null
    at /usr/src/app/node_modules/browsertime/lib/support/pathToFolder.js:40:37
    at Array.forEach (<anonymous>)
    at module.exports (/usr/src/app/node_modules/browsertime/lib/support/pathToFolder.js:39:18)
    at Collector.perIteration (/usr/src/app/node_modules/browsertime/lib/core/engine/collector.js:176:19)
    at Engine.runByScript (/usr/src/app/node_modules/browsertime/lib/core/engine/index.js:118:23)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-02-04 13:39:41] INFO: Running tests using Chrome - 3 iteration(s)
[2019-02-04 13:39:42] INFO: Start to measure };
[2019-02-04 13:39:42] ERROR: TypeError: Cannot read property 'replace' of null
    at /usr/src/app/node_modules/browsertime/lib/support/pathToFolder.js:40:37
    at Array.forEach (<anonymous>)
    at module.exports (/usr/src/app/node_modules/browsertime/lib/support/pathToFolder.js:39:18)
    at Collector.perIteration (/usr/src/app/node_modules/browsertime/lib/core/engine/collector.js:176:19)
    at Engine.runByScript (/usr/src/app/node_modules/browsertime/lib/core/engine/index.js:118:23)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-02-04 13:39:43] INFO: HTML stored in /sitespeed.io/sitespeed-result/module_exports/2019-02-04-13-39-39`

**Comment by soulgalore** [comment]
Yeah that's not the same as this issue. I think in some cases the click happened too fast so in 8.2 we wait a while before we start to wait on pageCompleteCheck, Can you try with that release? I got that test up and running continuously at dashboard.sitespeed.io and there it seems to work, but try with 8.2 hopefully that fixes things.

Best
Peter

**Comment by leathej1** [comment]
Sorry - I was not being clear in my bug report. This issue happens with *any* script I try to run, including a blank script. And I still am having the same issue in 8.2.0.

**Comment by soulgalore** [comment]
Aha! Running a blank script always generates that's error, that's why I didn't act on it. There's a couple of things you can do to help me debug, first just try a simple script and run it:

test.js
```
module.exports = async function(context, commands) {
  await commands.measure.start('https://www.sitespeed.io');
  await commands.measure.start('https://www.sitespeed.io/examples/');
  return commands.measure.start('https://www.sitespeed.io/documentation/');
};
```

`docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io test.js -n 1`

If that works, try one of your script that doesn't and either if you can run it locally on your desktop, then you can see on the browser window what happens. Another alternative is to change pageCompleteCheck (the script that waits on the page to finish loading). Easiest is to add `--spa ` to your run and watch what happens (then it will wait X seconds for an empty Resource Timing API).

If nothing of those works, please follow https://www.sitespeed.io/documentation/sitespeed.io/bug-report/#explain-how-to-reproduce-your-issue so I can try to reproduce it!

Thanks
Peter

**Comment by leathej1** [comment]
Test results:
```
uwadmins-MacBook-Pro:~ jamesleatherman$ docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io test.js -n 1
Google Chrome 72.0.3626.81 
Mozilla Firefox 65.0
[2019-02-04 21:16:04] INFO: Versions OS: linux 4.9.125-linuxkit nodejs: v10.13.0 sitespeed.io: 8.2.0 browsertime: 4.2.3 coach: 3.2.0
[2019-02-04 21:16:05] INFO: Running tests using Chrome - 1 iteration(s)
[2019-02-04 21:16:06] INFO: Start to measure module.exports
[2019-02-04 21:16:06] ERROR: TypeError: Cannot read property 'replace' of null
    at /usr/src/app/node_modules/browsertime/lib/support/pathToFolder.js:40:37
    at Array.forEach (<anonymous>)
    at module.exports (/usr/src/app/node_modules/browsertime/lib/support/pathToFolder.js:39:18)
    at Collector.perIteration (/usr/src/app/node_modules/browsertime/lib/core/engine/collector.js:176:19)
    at Engine.runByScript (/usr/src/app/node_modules/browsertime/lib/core/engine/index.js:143:23)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-02-04 21:16:06] INFO: Running tests using Chrome - 1 iteration(s)
[2019-02-04 21:16:07] INFO: Start to measure };
[2019-02-04 21:16:08] ERROR: TypeError: Cannot read property 'replace' of null
    at /usr/src/app/node_modules/browsertime/lib/support/pathToFolder.js:40:37
    at Array.forEach (<anonymous>)
    at module.exports (/usr/src/app/node_modules/browsertime/lib/support/pathToFolder.js:39:18)
    at Collector.perIteration (/usr/src/app/node_modules/browsertime/lib/core/engine/collector.js:176:19)
    at Engine.runByScript (/usr/src/app/node_modules/browsertime/lib/core/engine/index.js:143:23)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-02-04 21:16:09] INFO: HTML stored in /sitespeed.io/sitespeed-result/module_exports/2019-02-04-21-16-04```

**Comment by soulgalore** [comment]
Ah sorry my wrong in the example: If you run using sitespeed.io you need to use the `--multi `parameter see https://www.sitespeed.io/documentation/sitespeed.io/scripting/#run .
